benchmarks/yamauchi_takei_2016_anelasticity/anelasticity_temperature.cc

```diff
@@ -0,0 +1,469 @@
+				erf((log(peak_period/normalised_period))/(sqrt(2)*peak_width)))));
+		// determine J2 term (imaginary part of complex compliance)
+		loss_compliance=unrelaxed_compliance*(M_PI/2)*(background_amplitude*(pow(normalised_period,background_slope))+\
+					(peak_amplitude*exp(-1*(pow(log(peak_period/normalised_period),2)
+								/(2*pow(peak_width,2))))))+(unrelaxed_compliance*normalised_period);
+		// calculate anharmonic Vs
+		anharmonic_Vs=1/(sqrt(density*unrelaxed_compliance)*1e3);
+		// calculate Vs
+		anelastic_Vs=1/(sqrt(density*storage_compliance)*1e3);
+		// calculate attenuation
+		attenuation=loss_compliance/storage_compliance;
+	  return anelastic_Vs;
+	}
+	
+	// set up function to do Brent minimization
+    template <int dim>
+	double
+    AnelasticVs2T<dim>::
+	brent(std::function<double (double)> f, double lower_bound, double upper_bound, double TOL, double MAX_ITER)
+	{
+	    double a = lower_bound;
+	    double b = upper_bound;
+	    double fa = f(a);   // calculated now to save function calls
+	    double fb = f(b);   // calculated now to save function calls
+	    double fs = 0;      // initialize
+	    if (!(fa * fb < 0))
+	    {
+	        std::cout << "Signs of f(lower_bound) and f(upper_bound) must be opposites" << std::endl; // throws exception if root isn't bracketed
+	        return 0;
+	    }
+	    if (std::abs(fa) < std::abs(b)) // if magnitude of f(lower_bound) is less than magnitude of f(upper_bound)
+	    {
+	        std::swap(a,b);
+	        std::swap(fa,fb);
+	    }
+	    double c = a;           // c now equals the largest magnitude of the lower and upper bounds
+	    double fc = fa;         // precompute function evalutation for point c by assigning it the same value as fa
+	    bool mflag = true;      // boolean flag used to evaluate if statement later on
+	    double s = 0;           // Our Root that will be returned
+	    double d = 0;           // Only used if mflag is unset (mflag == false)
+	    for (unsigned int iter = 1; iter < MAX_ITER; ++iter)
+	    {
+	        // stop if converged on root or error is less than tolerance
+	        if (std::abs(b-a) < TOL)
+	        {
+	            return s;
+	        } // end if
+	        if (fa != fc && fb != fc)
+	        {
+	            // use inverse quadratic interpolation
+	            s =   ( a * fb * fc / ((fa - fb) * (fa - fc)) )
+	                + ( b * fa * fc / ((fb - fa) * (fb - fc)) )
+	                + ( c * fa * fb / ((fc - fa) * (fc - fb)) );
+	        }
+	        else
+	        {
+	            // secant method
+	            s = b - fb * (b - a) / (fb - fa);
+	        }
+	        /*
+	            crazy condition statement!:
+	            -------------------------------------------------------
+	            (condition 1) s is not between  (3a+b)/4  and b or
+	            (condition 2) (mflag is true and |s−b| ≥ |b−c|/2) or
+	            (condition 3) (mflag is false and |s−b| ≥ |c−d|/2) or
```

Comment:
is this code taken from somewhere?

Reply:
I've now replaced this Brent function with the boost version so this issue should now be fixed